src/apps/monitoring/kadiska/mode/alerts.pm

```diff
@@ -0,0 +1,229 @@
+                output_template => 'ok count: %s',
+                display_ok => 0,
+                closure_custom_perfdata => sub { return 0; }
+            }
+        },
+        { label => 'rule-warning-count',
+            type => 1,
+            set => {
+                key_values => [ { name => 'warning_count' } ],
+                output_template => 'warning count: %s',
+                display_ok => 0,
+                closure_custom_perfdata => sub { return 0; }
+            }
+        },
+        { label => 'rule-critical-count',
+            type => 1,
+            set => {
+                key_values => [ { name => 'critical_count' } ],
+                output_template => 'critical count: %s',
+                display_ok => 0,
+                closure_custom_perfdata => sub { return 0; }
+            }
+        },
+        { label => 'rule-nodata-count',
+            type => 1,
+            set => {
+                key_values => [ { name => 'nodata_count' } ],
+                output_template => 'no data count: %s',
+                display_ok => 0,
+                closure_custom_perfdata => sub { return 0; }
+            }
+        }
+    ];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        'filter-name:s'     => { name => 'filter_name' }
+    });
+
+    return $self;
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $raw_form_post = {
+        "select" => [
+            {
+                "rule_id:group" => "rule_id"
+            },
+            {
+                "rule_name:any" => ["any","rule_name"]
+            },
+            {
+                ["any","rule_name"] => ["any","rule_id"]
+            },
+            {
+                "critical:count" => ["sum","critical_count"]
+            },
+            {
+                "warning:count" => ["sum","warning_count"]
+            },
+            {
+                "ok:count" => ["sum","ok_count"]
+            },
+            {
+                "nodata:count" => ["sum","nodata_count"]
+            }
+        ],
+        "from" => "alert",
+        "groupby" => [
+            "rule_id:group"
+        ],
+        "orderby" => [
+            ["rule_name:any","desc"]
+        ],
+        "offset" => 0,
+        "limit" => 61
+    };
+
+    my $results = $options{custom}->request_api(
+        method => 'POST',
+        endpoint => 'query',
+        query_form_post => $raw_form_post
+    );
+
+    #Check if bad API request is submit
+    if (!exists $results->{data}) {
+        $self->{output}->add_option_msg(short_msg => 'No data result in API request.');
+        $self->{output}->option_exit();
+    }
+
+    $self->{total_critical}->{total_critical}=0;
+    $self->{total_warning}->{total_warning}=0;
+    $self->{total_nodata}->{total_nodata}=0;
```

Comment:
if think it's wrong. it should be:
```
$self->{global} = { total_critical => 0, total_warning => 0, total_nodata => 0 };
```

Reply:
Both seems work but I add your suggestion.